### ISSUE TITLE
Performance issue.

### ISSUE THREAD

**Issue by lanzarote0tr** [Dimigo-ieum/CLOVI#1]
(.train_venv) ethancho@MacBook-Pro-590 clovi % python train.py --use_fashion_mnist --epochs 5 Epoch 1/5 1969/1969 ━━━━━━━━━━━━━━━━━━━━ 0s 519ms/step - acc: 0.7749 - loss: 0.6431 Epoch 1: val_acc improved from -inf to 0.10157, saving model to export/ckpt_best.keras 1969/1969 ━━━━━━━━━━━━━━━━━━━━ 1073s 529ms/step - acc: 0.7749 - loss: 0.6431 - val_acc: 0.1016 - val_loss: 22.6486 Epoch 2/5 1969/1969 ━━━━━━━━━━━━━━━━━━━━ 0s 556ms/step - acc: 0.8249 - loss: 0.4781 Epoch 2: val_acc improved from 0.10157 to 0.10243, saving model to export/ckpt_best.keras 1969/1969 ━━━━━━━━━━━━━━━━━━━━ 1114s 566ms/step - acc: 0.8249 - loss: 0.4781 - val_acc: 0.1024 - val_loss: 10.8922 Epoch 3/5

I tried to train with train.py but the performance was so bad. I want to know what is going on.

**Comment by gil00** [comment]
I used below code and I got  acc 0.9 after 5Epochs
```
import numpy as np
import tensorflow as tf

# --- config ---
img_size = 28
batch_size = 32

# --- preprocess: add channel -> resize -> RGB -> [0,1] ---
def preprocess(img, label):
    img = tf.expand_dims(img, -1)                     # (28,28,1)
    img = tf.image.resize(img, (img_size, img_size))  # (H,W,1)
    img = tf.image.grayscale_to_rgb(img)              # (H,W,3)
    img = tf.cast(img, tf.float32) / 255.0            # [0,1]
    return img, label

# --- load data ---
(data_train, labels_train), (data_test, labels_test) = tf.keras.datasets.fashion_mnist.load_data()

# --- build datasets (map -> batch -> prefetch) ---
ds_train = (tf.data.Dataset.from_tensor_slices((data_train, labels_train))
            .shuffle(10000, reshuffle_each_iteration=True)
            .map(preprocess, num_parallel_calls=tf.data.AUTOTUNE)
            .batch(batch_size)
            .prefetch(tf.data.AUTOTUNE))

ds_val = (tf.data.Dataset.from_tensor_slices((data_test, labels_test))
          .map(preprocess, num_parallel_calls=tf.data.AUTOTUNE)
          .batch(batch_size)
          .prefetch(tf.data.AUTOTUNE))

# --- model ---
model = tf.keras.Sequential([
    tf.keras.layers.Input(shape=(img_size, img_size, 3)),
    tf.keras.layers.Conv2D(16, 3, activation='relu'),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(10, activation='softmax'),
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

model.fit(ds_train, validation_data=ds_val, epochs=5)
```

**Comment by gil00** [comment]
> (.train_venv) ethancho@MacBook-Pro-590 clovi % python train.py --use_fashion_mnist --epochs 5 Epoch 1/5 1969/1969 ━━━━━━━━━━━━━━━━━━━━ 0s 519ms/step - acc: 0.7749 - loss: 0.6431 Epoch 1: val_acc improved from -inf to 0.10157, saving model to export/ckpt_best.keras 1969/1969 ━━━━━━━━━━━━━━━━━━━━ 1073s 529ms/step - acc: 0.7749 - loss: 0.6431 - val_acc: 0.1016 - val_loss: 22.6486 Epoch 2/5 1969/1969 ━━━━━━━━━━━━━━━━━━━━ 0s 556ms/step - acc: 0.8249 - loss: 0.4781 Epoch 2: val_acc improved from 0.10157 to 0.10243, saving model to export/ckpt_best.keras 1969/1969 ━━━━━━━━━━━━━━━━━━━━ 1114s 566ms/step - acc: 0.8249 - loss: 0.4781 - val_acc: 0.1024 - val_loss: 10.8922 Epoch 3/5
> 
> I tried to train with train.py but the performance was so bad. I want to know what is going on.

val_acc : 0.1024 means real accuracy
acc : 0.8249 is train accuracy
so you might to try to get good val_acc.

I'll attach code and train weight on my repo link

git@github.com:gil00/clovi.git
